common/reducers/swap/index.ts

```diff
@@ -1,24 +1,14 @@
 import * as actionTypes from 'actions/swap';
 import { TypeKeys } from 'actions/swap/constants';
-import without from 'lodash/without';
-import {
-  buildDestinationAmount,
-  buildDestinationKind,
-  buildOriginKind
-} from './helpers';
-export const ALL_CRYPTO_KIND_OPTIONS = ['BTC', 'ETH', 'REP'];
-const DEFAULT_ORIGIN_KIND = 'BTC';
-const DEFAULT_DESTINATION_KIND = 'ETH';
+import { normalize } from 'normalizr';
+import * as schema from './schema';
 
 export interface State {
-  originAmount: number | null;
-  destinationAmount: number | null;
-  originKind: string;
-  destinationKind: string;
-  destinationKindOptions: string[];
-  originKindOptions: string[];
   step: number;
-  bityRates: any;
+  origin: actionTypes.SwapInput | null;
```

Comment:
Action Creator Types should follow convention here: https://github.com/MyEtherWallet/MyEtherWallet#actiontypests